ebusd-2.1.x/de/tem/_templates.csv

```diff
@@ -0,0 +1,54 @@
+#name,type/templates,divider/values,unit,comment
+# Bartl TEM,,,,
+type,UCH,0x00=none;0x02=on-off;0x04=Zeit;0x09=selection;0x0d=Temperatur;0x40=Dauer;0x4d=TemperaturSchwelle;0x8d=TemperaturSoll;0xc0=range,,Datentyp
+unit,UCH,0x00=none;0x02=°C;0x04=K;0x08=%;0x0a=h;0x0c=min;0x10=kW;0x2a=hh:mm,,Einheit
+maxs:max,SIN,,,Maximum
+mins:min,SIN,,,Minimum
+maxs_10:max,SIN,10,,Maximum (Teiler 10)
+mins_10:min,SIN,10,,Minimum (Teiler 10)
+wd,BDY,,,Weekday
+range,SCH,,,Range
+command,TEM_P;type;unit,,,Kommando
+temp1:temp,SIN,,°C,Temperatur
+temp1_1b:temp,SCH;IGN:1,,°C,Temperatur
+temp10:temp,SIN,10,°C,Temperatur (Teiler 10)
+temp10K:temp,SIN,10,K,Temperatur (Teiler 10K)
+temp2:temp,SIN,2,°C,Temperatur (Teiler 2)
+temp256:temp,D2B,,°C,Temperatur (Teiler 256)
+max_temp1:max,temp1,,,Maximum
+min_temp1:min,temp1,,,Minimum
+max_temp1_1b:max,SCH;IGN:1,,,Maximum
+min_temp1_1b:min,SCH;IGN:1,,,Minimum
+max_temp10:max,temp10,,,Maximum
+min_temp10:min,temp10,,,Minimum
+max_temp10K:max,temp10K,,,Maximum
+min_temp10K:min,temp10K,,,Minimum
+max_temp2:max,temp2,,,Maximum
+min_temp2:min,temp2,,,Minimum
+max_time:max,MIN,,,Maximum
+min_time:min,MIN,,,Minimum
+max_wd:max,wd;IGN:1,,,Maximum
+min_wd:min,wd;IGN:1,,,Minimum
+max_range:max,range;IGN:1,,,Maximum
+min_range:max,range;IGN:1,,,Minimum
+parameter1,command;maxs;mins,,,Parameter
+parameter1_10,command;maxs_10;mins_10,,,Parameter10
+parameter2,command;UCH;UCH,,,Parameter
+parameter_temp1,command;max_temp1;min_temp1,,,ParameterTemp1
+parameter_temp1_1b,command;max_temp1_1b;min_temp1_1b,,,ParameterTemp1
+parameter_temp10,command;max_temp10;min_temp10,,,ParameterTemp10
+parameter_temp10K,command;max_temp10K;min_temp10K,,,ParameterTemp10K
+parameter_temp2,command;max_temp2;min_temp2,,,ParameterTemp2
+parameter_time,command;max_time;min_time,,,ParameterTime
+parameter_wd,command;max_wd;min_wd,,,ParameterWeekday
+parameter_range,command;max_range;min_range,,,ParameterRange
+btime,BTI,,,Binary Time
+bdate,BDA,,,Binary Date
+status,SIN,,,Status
+hcd100,HCD,100,h,Betriebsstunden (Teiler 10)
+tp_2h,BI0;BI1;BI2;BI3;BI4;BI5;BI6;BI7,,,Zeitprogramm 2 Stunden
+tp_8h,tp_2h;tp_2h;tp_2h;tp_2h,,,Zeitprogramm 8 Stunden
+tp_20h,tp_2h;tp_2h;tp_2h;tp_2h;tp_2h;tp_2h;tp_2h;tp_2h;tp_2h;tp_2h,,,Zeitprogramm 20 Stunden
+tp_mode,UCH;UCH;UCH;UCH;UCH;UCH;UCH;UCH;UCH;IGN:1,,,Modus Zeitprogramm
+tp_week,tp_20h;tp_20h;tp_20h;tp_20h;tp_20h;tp_20h;tp_20h;tp_20h;tp_8h,,,Zeitprogramm Woche
```

Comment:
are you sure about this one and tp_all? it would mean that the message NN is bigger than 160 bytes which is way out of the ebus specs. maybe you could submit an example hex sequence?

Reply:
I will check that these days...

Reply:
tp_all and tp_week is used as a template for chained messages, for example: https://github.com/john30/ebusd-configuration/pull/129/files#diff-a1f7f271448f1fabf38cf161bfb613d9320f5662d424188229934fa4459ed97fR135

`ebusctl read -f -c ww_timeprogramm tp_ww` just triggers the single tp*_ww and chains them for me...

Reply:
`311506220405bd004e / 0a00010000000000000000 = 1: ww_timeprogramm tp0_ww
311506220405bd014e / 0a0000e0ffffffffffffff = 1: ww_timeprogramm tp1_ww
311506220405bd024e / 0aff000000e0ffffffffff = 1: ww_timeprogramm tp2_ww
311506220405bd034e / 0affffff000000e0ffffff = 1: ww_timeprogramm tp3_ww
311506220405bd044e / 0affffffffff000000e0ff = 1: ww_timeprogramm tp4_ww
311506220405bd054e / 0affffffffffffff000000 = 1: ww_timeprogramm tp5_ww
311506220405bd064e / 0ae0ffffffffffffffff00 = 1: ww_timeprogramm tp6_ww
311506220405bd074e / 0a0000e0ffffffffffffff = 1: ww_timeprogramm tp7_ww
311506220405bd084e / 0aff000000e0ffffffffff = 1: ww_timeprogramm tp8_ww
311506220405bd094e / 04ffffff00 = 1: ww_timeprogramm tp9_ww`

Reply:
which is
```
andreas@orangepiplus:~$ ebusctl read -f -c ww_timeprogramm tp_ww
0;1;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;0;0;0;0;0;0;0;0
```

Reply:
ah, didn't see the chain, that's clear then, thanks